backend/app/comment_handler.go

```diff
@@ -0,0 +1,168 @@
+package app
+
+import (
+	"errors"
+	"log"
+	"net/http"
+
+	"gorm.io/gorm"
+
+	"github.com/codescalersinternships/Flyspray/models"
+	"github.com/gin-gonic/gin"
+)
+
+// Response is a struct that holds the ok response
+type Response struct {
+	Message string           `json:"message"`
+	Payload []models.Comment `json:"payload"`
+}
+
+// ErrorResponse is a struct that holds the error response
+type ErrorResponse struct {
+	Error string `json:"error"`
+}
+
+// CreateComment creates a comment on a specific bug
+func (app *App) CreateComment(c *gin.Context) {
+
+	comment := models.Comment{}
+
+	if err := c.ShouldBindJSON(&comment); err != nil {
+		log.Println("failed to create the comment:", err)
+		c.JSON(http.StatusBadRequest,
+			ErrorResponse{Error: "bugID,ownerID and summary of the comment should be provided "})
+		return
+	}
+
+	if err := comment.Validate(); err != nil {
+		log.Println("bugID,ownerID and summary of the comment should be validated", err)
+		c.JSON(http.StatusBadRequest,
+			ErrorResponse{Error: "comment is not validated"})
+		return
+	}
+
+	comment, err := app.client.CreateComment(comment)
+	if err != nil {
+		log.Fatal(err)
+		c.JSON(http.StatusInternalServerError, ErrorResponse{Error: "failed to create comment"})
+		return
+	}
+
+	c.JSON(http.StatusCreated,
+		Response{Message: "comment is created",
+			Payload: []models.Comment{comment}})
+
+}
+
+// GetComment gets a comment on a specific bug by id
+func (app *App) GetComment(c *gin.Context) {
+
+	id := c.Param("id")
+
+	if id == "" {
+		log.Println("comment ID is required")
+		c.JSON(http.StatusNotFound, ErrorResponse{Error: "comment ID is required"})
+		return
+	}
+
+	comment, err := app.client.GetComment(id)
+	if err != nil {
+		log.Println("comment is not found:", err)
+		c.JSON(http.StatusNotFound,
+			ErrorResponse{Error: "comment is not found"})
+		return
+
+	}
+
+	c.JSON(http.StatusOK,
+		Response{Message: "comment is found",
+			Payload: []models.Comment{comment}})
+
+}
+
+// DeleteComment deletes a comment on a specific bug by id
+func (app *App) DeleteComment(c *gin.Context) {
+
+	id := c.Param("id")
+
+	if id == "" {
+		log.Println("comment ID is required")
+		c.JSON(http.StatusBadRequest, ErrorResponse{Error: "comment ID is required"})
+		return
+	}
+
+	if err := app.client.DeleteComment(id); err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			log.Println("comment is not found:", err)
+			c.JSON(http.StatusNotFound, ErrorResponse{Error: "comment is not found"})
+		} else {
```

Comment:
it should be a check `if errors.Is(err, gorm.ErrRecordNotFound)` then `if err == nil`